### DISCUSSION TITLE
`npx cap sync` overwrites changes in `Pods-App.debug|release`

### DISCUSSION THREAD

**Discussion by frederikheld** [ionic-team/capacitor#7487] [Q&A]
Hey everyone,

I had to set `ENABLE_USER_SCRIPT_SANDBOXING = NO` in `Pods-App.debug` and `Pods-App.release`. Unfortunately this change gets overwritten every time I run `npx cap sync`. Is there a way to set this value in a persistent way?

Thanks for your help!
Frederik

**Comment by jcesarmobile** [comment]
It's not Capacitor but CocoaPods itself, you should never manually edit Pods files, CocoaPods will reset those files in `pod update`, which Capacitor runs on `npx cap sync`.

For making permanent changes in the Pods projects you have to edit your `Podfile` so it makes the changes, something like
```
post_install do |installer|
  installer.generated_projects.each do |project|
    project.targets.each do |target|
      target.build_configurations.each do |config|
        config.build_settings["ENABLE_USER_SCRIPT_SANDBOXING"] = "NO"
      end
    end
  end
end
```

**Reply by frederikheld** [reply]
Thanks for the clarification!

I found out that Xcode 15 sets this value to "YES" in `/ios/App/App.xcodeproj/project.pbxproj` where it can be set back to "NO". This seems like the more straightforward solution. But is this file safe to edit?

Follow up question: this "hack" is only necessary because some Capacitor script magic seems to be faulty. With sandboxing enabled (which seems to be the default since Xcode 15), the build throws the following error:

```
Sandbox: bash(77645) deny(1) file-read-data [...]/app/ios/App/Pods/Target Support Files/Pods-App/Pods-App-frameworks.sh
```

My research revealed, that with sandboxing enabled, I need to specify input/output files for scripts ([source](https://stackoverflow.com/a/78472315/10043870)).

I found the file but I have no idea what is going on there and how to fix it. Could you help me?

**Reply by jcesarmobile** [reply]
Yes, editing your project files should be safe, Capacitor never modifies your project after creation. As long as you edit your project files and not CocoaPods files you should be good.

Again, you are referring to CocoaPods files when you say "Capacitor script magic seems to be faulty".

Capacitor 6 has experimental support for Swift Package Manager, you can try to use that instead of CocoaPods. 

https://capacitorjs.com/docs/ios/spm

**Reply by frederikheld** [reply]
> Again, you are referring to CocoaPods files when you say "Capacitor script magic seems to be faulty".

Everything inside the `ios` directory (including the CocoaPods configuration) is created by Capacitor in the first place (that's the Capacitor script magic that I refer to). So I would expect it to take care of that issue. Having sandboxing enabled by default was introduced for a reason, disabling it is not a solution, just a hack.

Thanks for hinting me to SPM. Is SPM be expected to replace CocoaPods in the long run?

**Comment by SaebAmini** [comment]
I found that during an Xcode operation `ENABLE_USER_SCRIPT_SANDBOXING = YES` was set in the `pbxproj` file which was causing this issue. Manually editing the value  to `NO` fixed it and it hasn't come back.

**Comment by essmos** [comment]
make file scripts/fix-sandboxing.rb

require 'xcodeproj'

project_path = 'ios/App/App.xcodeproj'
project = Xcodeproj::Project.open(project_path)

project.build_configurations.each do |config|
  config.build_settings['ENABLE_USER_SCRIPT_SANDBOXING'] = 'NO'
end

project.targets.each do |target|
  target.build_configurations.each do |config|
    config.build_settings['ENABLE_USER_SCRIPT_SANDBOXING'] = 'NO'
  end
end

project.save

add to package.json in section scripts
"postsync": "ruby scripts/fix-sandboxing.rb"

run : npx cap sync && npm run postsync
